File: frontend/src/services/articleService.js
import axiosClient from "./axiosClient";

export const createArticle = (payload) =>
  axiosClient.post("/articles", payload);

export const updateArticle = (articleId, payload) =>
  axiosClient.put(`/articles/${articleId}`, payload);

export const getArticles = (params) =>
  axiosClient.get("/articles", { params });

export const getArticleById = (articleId) =>
  axiosClient.get(`/articles/${articleId}`);

export const getArticleBySlug = (slug) =>
  axiosClient.get(`/articles/slug/${slug}`);

export const publishArticle = (articleId) =>
  axiosClient.patch(`/articles/${articleId}/publish`);

export const deleteArticle = (articleId) =>
  axiosClient.delete(`/articles/${articleId}`);

File: backend/src/main/java/com/clinicmanagement/article/ArticleService.java
package com.clinicmanagement.article;

import com.clinicmanagement.article.dto.ArticleRequest;
import com.clinicmanagement.article.dto.ArticleResponse;
import com.clinicmanagement.common.dto.PageResponse;
import org.springframework.data.domain.Pageable;

public interface ArticleService {
    ArticleResponse createArticle(Long userId, ArticleRequest request);
    ArticleResponse updateArticle(Long articleId, Long userId, boolean isAdmin, ArticleRequest request);
    ArticleResponse getArticleById(Long articleId);
    ArticleResponse getArticleBySlug(String slug);
    PageResponse<ArticleResponse> getAllArticles(String status, Pageable pageable);
    void deleteArticle(Long articleId);
    ArticleResponse publishArticle(Long articleId);
}
